src/transport/connection/tests.rs

```diff
@@ -0,0 +1,272 @@
+                    break;
+                }
+                Ok(_) => {
+                    panic!("Mock Server received more client requests than declared");
+                }
+                Err(e) => {
+                    panic!("Mock Server stream errored during keep_alive: {}", e);
+                }
+            };
+        }
+    }
+}
+
+struct MockServer {
+    listener: TcpListener,
+}
+
+impl MockServer {
+    fn new(listener: TcpListener) -> Self {
+        Self { listener }
+    }
+
+    fn process(&self, mut events: VecDeque<Event>) {
+        loop {
+            if events.is_empty() {
+                break;
+            }
+
+            let mut messages = VecDeque::new();
+            while let Some(event) = events.pop_front() {
+                match event {
+                    Event::Message(message) => messages.push_back(message),
+                    Event::Restart => break,
+                }
+            }
+
+            MockStream::new(self.listener.accept().unwrap().0, messages, events.is_empty()).process();
+        }
+    }
+}
+
+struct TestServer {
+    listener: TcpListener,
+    handle: JoinHandle<()>,
+}
+impl TestServer {
+    fn start(events: Vec<Event>) -> Self {
+        let listener = TcpListener::bind("127.0.0.1:0").unwrap();
+
+        let listener_clone = listener.try_clone().unwrap();
+
+        let handle = std::thread::spawn(move || {
+            let server = MockServer::new(listener_clone);
+            server.process(VecDeque::from(events));
+        });
+        Self { listener, handle }
+    }
+    fn address(&self) -> SocketAddr {
+        self.listener.local_addr().unwrap()
+    }
+}
+
+const AAPL_CONTRACT: &str  = "AAPL|STK||0||SMART|USD|AAPL|NMS|NMS|265598|0.01||ACTIVETIM,AD,ADDONT,ADJUST,ALERT,ALGO,ALLOC,AON,AVGCOST,BASKET,BENCHPX,CASHQTY,COND,CONDORDER,DARKONLY,DARKPOLL,DAY,DEACT,DEACTDIS,DEACTEOD,DIS,DUR,GAT,GTC,GTD,GTT,HID,IBKRATS,ICE,IMB,IOC,LIT,LMT,LOC,MIDPX,MIT,MKT,MOC,MTL,NGCOMB,NODARK,NONALGO,OCA,OPG,OPGREROUT,PEGBENCH,PEGMID,POSTATS,POSTONLY,PREOPGRTH,PRICECHK,REL,REL2MID,RELPCTOFS,RPI,RTH,SCALE,SCALEODD,SCALERST,SIZECHK,SMARTSTG,SNAPMID,SNAPMKT,SNAPREL,STP,STPLMT,SWEEP,TRAIL,TRAILLIT,TRAILLMT,TRAILMIT,WHATIF|SMART,AMEX,NYSE,CBOE,PHLX,ISE,CHX,ARCA,NASDAQ,DRCTEDGE,BEX,BATS,EDGEA,BYX,IEX,EDGX,FOXRIVER,PEARL,NYSENAT,LTSE,MEMX,IBEOS,OVERNIGHT,TPLUS0,PSX|1|0|APPLE INC|NASDAQ||Technology|Computers|Computers|US/Eastern|20250324:0400-20250324:2000;20250325:0400-20250325:2000;20250326:0400-20250326:2000;20250327:0400-20250327:2000;20250328:0400-20250328:2000|20250324:0930-20250324:1600;20250325:0930-20250325:1600;20250326:0930-20250326:1600;20250327:0930-20250327:1600;20250328:0930-20250328:1600|||1|ISIN|US0378331005|1|||26,26,26,26,26,26,26,26,26,26,26,26,26,26,26,26,26,26,26,26,26,26,26,26,26||COMMON|0.0001|0.0001|100|";
+
+// Examples:
+// Testing connect -> request -> disconnect -> reconnect -> request
+#[test]
+fn test_mock_connection() {
```

Comment:
I think this will be easier to reason about if broken into 3 tests.

* **test_establish_connection** - would test just handshake and start api
* **test_send_request** - test a single request (e.g. connect and get  contract details)
* **test_reconnect** - test ability to reconnect. simplest way to do this is with an API like `Client::managed_accounts`. I haven't gone back and added this to every API that could support this but some like `Client::managed_accounts` can automatically retry after a reconnect.